snooty/test_parser.py

```diff
@@ -96,12 +97,21 @@ def test_tabs() -> None:
             </directive>
         </directive>
 
+        <directive name="tabs">
+            <directive name="tab" tabid="http">
+                <text>HTTP</text>
+                <paragraph><text>Tab Content</text></paragraph>
+            </directive>
+            <directive name="tip">
+                <text>This isn't a tab</text>
+            </directive>
+        </directive>
```

Comment:
This behavior seems suspect but I was replicating what I *thought* was the existing behavior for raising a TabMustBeDirective diagnostics error.

Reply:
Interesting, yeah. I think the desired outcome would be to omit the invalid child entirely (though correct me if I'm wrong, @i80and). I would probably prefer to see the `tip` node omitted here.

Reply:
I would also strongly prefer to omit the invalid node :D

There's a good argument to be made that we shouldn't be mutating node structure to that extent during AST construction, but until we have a separate pass available, it's fine.